### PR TITLE
Product creation implementation

### PR DESCRIPTION
**Pull request by mayerbryan**
code implementation for the product creation functionality

### CODE REVIEW

File: Controllers/HomeController.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+using ProductsCatalog.Models;
+
+namespace ProductsCatalog.Controllers{
+    [ApiController]
+    
+    public class HomeController : ControllerBase {
+       
+              
+
+        [HttpPost("/")]
```
Comment (btalanski) [inline Controllers/HomeController.cs]: APIs need to follow a URL scheme. Usually the pattern is: `<api version>/<resource>/<>`

So if you need to create a Product, the url would be something like: `v1/products`

More info:
- https://hevodata.com/learn/rest-api-best-practices/
- https://www.thecodebuzz.com/restful-api-url-naming-conventions-best-practices/
- https://www.altexsoft.com/blog/rest-api-design/

File: Controllers/HomeController.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+using Microsoft.AspNetCore.Mvc;
+using ProductsCatalog.Data;
+using ProductsCatalog.Models;
+
+namespace ProductsCatalog.Controllers{
+    [ApiController]
+    
+    public class HomeController : ControllerBase {
```
Comment (btalanski) [inline Controllers/HomeController.cs]: I suggest the controller name to be the same as the API's resource name: `ProdutcsController`

**Comment by btalanski** [comment]
I believe the folders `obj` and `bin` should not be committed to github. Those files contain references to the user's file system, such as `c:\Users\Mayer\Documents\Dev\Projects\ProductsCatalog\` .

Probably these will cause issues if another user tries to clone and run your application in their machines, as the paths won't be the same. Usually a `.gitignore` file needs to be added to the root of the project to ensure files only stay in your local machine and don't get pushed to source control.

Take a look at this page: https://elanderson.net/2020/10/add-git-ignore-to-an-existing-visual-studio-solution-new-git-experience/ 

If it doesn't work, try the solution below:

Delete those folders and their content, then run `dotnet new gitignore` to generate the ignore file. Push the changes to github and ensure those folders are deleted. Probably when you build the application locally again, it should regenerate those files, but `.gitignore` will ensure they are not added to the repository again.

File: Controllers/HomeController.cs (hunk 1 above)
Comment (mayerbryan) [reply Controllers/HomeController.cs]: I tried to implement this but the way he show in the course was not working when I use the v1 route in the begining of the class, I will try to fix this problem and implement.

File: Controllers/HomeController.cs (hunk 2 above)
Comment (mayerbryan) [reply Controllers/HomeController.cs]: just to understand, we dont need a home controller? can we have multiple controllers in one api? if yes I understand why the name change..

File: Controllers/HomeController.cs (hunk 1 above)
Comment (btalanski) [reply Controllers/HomeController.cs]: Can't you just change `[HttpPost("/")]` to `[HttpPost("/v1/products")]` and recompile ?

File: Controllers/HomeController.cs (hunk 2 above)
Comment (btalanski) [reply Controllers/HomeController.cs]: Yes, an application can have multiple controllers. Usually you will have a controller for each type of resource.

**Review by btalanski — COMMENTED** [review]
Approved

**Pull request merged**